qiskit/circuit/quantumcircuit.py

```diff
@@ -162,18 +162,16 @@ def __init__(self, *regs, name=None, global_phase=0, metadata=None):
                                    "provided)" % ([type(reg).__name__ for reg in regs], regs))
 
             regs = tuple(int(reg) for reg in regs)  # cast to int
-
+        self._base_name = None
         if name is None:
-            name = self.cls_prefix() + str(self.cls_instances())
-            if sys.platform != "win32" and not is_main_process():
-                name += '-{}'.format(mp.current_process().pid)
-        self._increment_instances()
-
-        if not isinstance(name, str):
+            self._base_name = self.cls_prefix()
+            self._name_update()
+        elif not isinstance(name, str):
             raise CircuitError("The circuit name should be a string "
                                "(or None to auto-generate a name).")
-
-        self.name = name
+        else:
+            self.name = name
```

Comment:
Should we also set `self._base_name = name` here (or how to have `_name_update` know about a user-provided name)?

Reply:
yep, otherwise we get circuits named "None"